src/index.ts

```diff
@@ -102,37 +100,41 @@ function convertBundleIntoReact(
     );
   }
 
-  const content = bundle.content
-    .split(/(\s+)/)
-    .reduce((words: React.ReactNode[], word: string, index: number) => {
-      // If this is a separator, re-add the space removed from split.
-      if (index % 2 === 1) {
-        words.push(word);
-        return words;
-      }
-
-      // If  this isn't a link, just return the word as-is.
-      if (!LINK_REGEX.test(word)) {
-        words.push(word);
-        return words;
-      }
-
-      // Make sure the href we generate from the link is fully qualified. We assume http
-      // if it starts with a www because many sites don't support https
-      const href = word.startsWith("www.") ? `http://${word}` : word;
-      words.push(
-        React.createElement(
-          "a",
-          {
-            key: index,
-            href,
-            target: "_blank"
-          },
-          `${word}`
-        )
-      );
-      return words;
-    }, [] as React.ReactNode[]);
+  const content: React.ReactNode[] = [];
+  const linkRegex = /(\s+|^)(https?:\/\/(?:www\.|(?!www))[^\s.]+\.[^\s]{2,}|www\.[^\s]+\.[^\s]{2,})(\s+|$)/g;
```

Comment:
Nit: Let's use named groups here since we are extracted matched groups from the regular expression more often throughout the code now.

Reply:
Support for named capture groups is still [a tad low](https://caniuse.com/mdn-javascript_builtins_regexp_named_capture_groups), so I just refactored it to be more readable for now.

If you're not worried about the compatibility there just let me know and I'll update it to use named groups.

Reply:
89% usage is reasonable but I'm fine with this approach. Thanks for striking a balance here.